python/grass/temporal/register.py

```diff
@@ -31,6 +31,44 @@
 ###############################################################################
 
 
+def build_abstract_map_dataset(map_item, element):
+    """Build AbstractMapDataset from map item and element type.
+
+    :param map_item: The id of the map to build an abstract dataset for.
+                     Can be of the form: mapname, mapname@mapsetname, or
+                     mapname:layer / mapname:layer@mapsetname for vector
+                     maps
+    :param element: The mapset element of the maps: cell, grid3d or vector
+    :return: ID String build with the AbstractMapDataset class
+
+    .. code-block:: python
+
+        >>> import grass.temporal as tgis
+        >>> tgis.register.build_abstract_map_dataset("vectormap:1@PERMANENT", "vector")
+        'vectormap:1@PERMANENT'
+        >>> tgis.register.build_abstract_map_dataset("rastermap@PERMANENT", "cell")
+        'rastermap@PERMANENT'
+        >>> tgis.register.build_abstract_map_dataset("raster3dmap@PERMANENT", "grid3d")
+        'raster3dmap@PERMANENT'
+
+    """
+    map_name, map_mapset = (
+        map_item.split("@")[0:2] if "@" in map_item else (map_item, None)
+    )
+    map_name, map_layer = map_name.split(":") if ":" in map_name else (map_name, None)
```

Comment:
In both cases, you can limit the number of splits which should be performed. The first call will be simpler and faster, the second one faster and safer.

Reply:
Addressed in the `build_id_from_search_path` function